datafusion/sql/Cargo.toml

```diff
@@ -54,6 +54,7 @@ strum = { version = "0.26.1", features = ["derive"] }
 [dev-dependencies]
 ctor = { workspace = true }
 datafusion-functions = { workspace = true, default-features = true }
+datafusion-functions-aggregate = { workspace = true }
```

Comment:
imported for test

Reply:
I think this is ok.

We have had bad experiences in the past with making semi-circular dependencies for tests (as I recall it messes up the release process to crates.io. THough maybe this one is ok, because datafusion-functions-aggregate doesn't depend on datafusion-sql either directly or transitively

If we are going to go with this approach, perhaps we could remove the stubs added in https://github.com/apache/datafusion/pull/10816 🤔